upup/pkg/fi/cloudup/terraform/target_hcl2.go

```diff
@@ -122,7 +122,7 @@ func (t *TerraformTarget) writeProviders(buf *bytes.Buffer) {
 	if t.Cloud.ProviderID() == kops.CloudProviderGCE {
 		providerBody["project"] = t.Project
 	}
-	if t.Cloud.ProviderID() != kops.CloudProviderHetzner {
+	if t.Cloud.ProviderID() != kops.CloudProviderHetzner && t.Cloud.ProviderID() != kops.CloudProviderDO {
```

Comment:
I understand that we'd set the region here unless the cloud provider is Hetzner or DO. To confirm, is that the desired behavior? DO does support regions, right?

Reply:
Yes that's correct. Per DO Terraform provider [reference here](https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs#argument-reference), it doesn't have the region attribute specified.
Though, we fetch the spaces region from the `S3_ENDPOINT` environment variable. And the droplet and volume region is associated to the terraform resource based on the `region` attribute as [referenced here.](https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/data-sources/droplet#region)